app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -25,18 +25,30 @@ def create_sub_plans
         job_invocation = JobInvocation.find(input[:job_invocation_id])
         proxy_selector = RemoteExecutionProxySelector.new
 
-        # composer creates just "pattern" for template_invocations because target is evaluated
-        # during actual run (here) so we build template invocations from these patterns
-        job_invocation.targeting.hosts.map do |host|
+        ::Host.where(:id => current_batch).map do |host|
```

Comment:
I'm afraid we do. The host object is passed to the `RunHostJob` action where the permissions are verified for it.

I'm planning to look into permissions in rex later today, so I'll file another issue and open a PR if I find a better way to do it since it's not really in the scope of this issue.